crates/cast/src/main.rs

```diff
@@ -111,25 +121,44 @@ fn main() -> Result<()> {
         ValueFormat::Default
     };
 
-    let mut config = parse_scarb_config(&cli.profile, &cli.path_to_scarb_toml)?;
+    which::which("scarb")
+        .context("Cannot find `scarb` binary in PATH. Make sure you have Scarb installed https://github.com/software-mansion/scarb")?;
+
+    let package_data = get_package_data(&cli);
+    let mut config = match &package_data.package {
+        Ok(package) => get_scarb_config(&cli.profile, Some(package))?,
+        Err(_) => get_scarb_config(&cli.profile, None)?,
+    };
+
     update_cast_config(&mut config, &cli);
 
     let provider = get_provider(&config.rpc_url)?;
     let runtime = Runtime::new().expect("Could not instantiate Runtime");
 
     if let Commands::Script(script) = cli.command {
+        package_data.manifest_path?;
```

Comment:
I think this line is confusing

Reply:
It pops the first error it encounters, not sure how to do it better I think it makes sense as a missing path is definitely a root cause of missing metadata